python/cuda_parallel/cuda/parallel/experimental/algorithms/merge_sort.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) 2024-2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import ctypes
+from typing import Callable
+
+import numba
+from numba import cuda
+from numba.cuda.cudadrv import enums
+
+from .. import _cccl as cccl
+from .._bindings import get_bindings, get_paths
+from .._utils import protocols
+from ..iterators._iterators import IteratorBase
+from ..typing import DeviceArrayLike
+
+
+def _dtype_validation(dt1, dt2):
```

Comment:
@shwina's PR #3718 removed this for reduce to resolve performance issues. I'm guessing we'll have the same issues here. Probably it's best to omit this dtype validation from the start for merge_sort?

Reply:
Removed and also updated the code to follow the new patterns more closely

Reply:
I still see the _dtype_validation() function here, although it isn't called anymore. Oversight?